### PR TITLE
[WIP][Enhancement] Support updating lake compation task queue count at run…

### PR DESCRIPTION
**Pull request by wxl24life**
…time

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 36 / 43 (83.72%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|5|00.00%|[155, 156, 157, 158, 159]|
|:large_blue_circle:|src/storage/lake/compaction_scheduler.cpp|32|34|94.12%|[164, 165]|
|:large_blue_circle:|src/storage/lake/compaction_scheduler.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 41 / 54 (75.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/http/action/update_config_action.cpp|0|5|00.00%|[155, 156, 157, 158, 159]|
|:large_blue_circle:|src/storage/lake/compaction_scheduler.cpp|37|45|82.22%|[164, 165, 176, 220, 221, 222, 223, 225]|
|:large_blue_circle:|src/storage/lake/compaction_scheduler.h|4|4|100.00%|[]|

**Comment by wxl24life** [comment]
> Could you please add some unit tests?

yes, i will do it later

File: be/src/common/config.h [hunk 1]
```diff
@@ -146,7 +146,7 @@ CONF_Int32(sleep_one_second, "1");
 CONF_Int32(sleep_five_seconds, "5");
 
 // The count of thread to compact
-CONF_Int32(compact_threads, "4");
+CONF_mInt32(compact_task_queue_count, "4");
```
Comment (sduzh) [inline be/src/common/config.h:149]: It looks like this configuration is only used in the shared data mode, how about standardizing the naming style with the other configurations and renaming it `lake_compactiton_max_concurrency`?

File: be/src/common/config.h [hunk 2]
```diff
@@ -146,7 +146,7 @@ CONF_Int32(sleep_one_second, "1");
...
 CONF_Int32(compact_thread_pool_queue_size, "100");
```
Comment (sduzh) [inline be/src/common/config.h:150]: This configuration is unused now, could you please remove it by the way?

File: be/src/storage/lake/compaction_scheduler.h [hunk 3]
```diff
@@ -196,16 +201,19 @@ class CompactionScheduler {
...
     Limiter _limiter;
     bthread::Mutex _contexts_lock;
     butil::LinkedList<CompactionTaskContext> _contexts;
-    int _task_queue_count;
-    TaskQueue* _task_queues;
+
+    std::atomic<int> _task_queue_count;
+    // use a mutex to protect the resize operation of _task_queues at runtime
+    std::mutex _task_queues_mutex;
+    std::vector<TaskQueue> _task_queues;
```
Comment (sduzh) [inline be/src/storage/lake/compaction_scheduler.h:216]: How about wrapping these fields together in a utility class that provides queue-like APIs?

File: be/src/storage/lake/compaction_scheduler.cpp [hunk 4]
```diff
@@ -159,14 +205,35 @@ void CompactionScheduler::steal_task(int start_index, std::unique_ptr<Compaction
...
+            if (!ret) {
+                // no remaining tasks in `task_queues[id]`
+                return;
+            }
+            auto idx = choose_task_queue_by_txn_id(context->txn_id);
+            {
+                std::lock_guard<std::mutex> lock(_task_queues_mutex);
+                // re-schedule the compaction task
+                _task_queues[idx].put(std::move(context));
+            }
```
Comment (sduzh) [inline be/src/storage/lake/compaction_scheduler.cpp:223]: Should it `continue` here instead of going to the processing below?

**Pull request closed without merging**